ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/show/ShowPartitionsOperation.java

```diff
@@ -54,12 +54,12 @@ public ShowPartitionsOperation(DDLOperationContext context, ShowPartitionsDesc d
   @Override
   public int execute() throws HiveException {
     Table tbl = context.getDb().getTable(desc.getTabName());
-    if (!tbl.isPartitioned()) {
```

Comment:
why did we move this? Doesn't it work for iceberg tables, we should have a partition spec in HMS?

Reply:
tbl.isPartitioned() evaluates to false for iceberg tables

Reply:
we do have part spec in HMS, so why do we return false for Iceberg?

Reply:
https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/metadata/Table.java#L612

As discussed,  List<FieldSchema> partKeys = tTable.getPartitionKeys();  returns null for iceberg.

I will raise a follow up jira to address this.